persistence/src/commonTest/kotlin/com/wire/kalium/persistence/dao/ConversationDAOTest.kt

```diff
@@ -296,6 +317,210 @@ class ConversationDAOTest : BaseDatabaseTest() {
+            newConversationEntity(
+                id = QualifiedIDEntity("3", "someDomain"),
+                lastReadDate = "2000-01-01T12:30:00.000Z",
+                lastModified = "2000-01-01T12:00:00.000Z"
+            )
+        )
+
+    }
+
+    @Test
+    fun givenConversationsHaveLastReadDateAfterModified_whenGettingUnReadConversationCount_ThenReturnTheExpectedCount() = runTest {
+        // given
+        conversationDAO.insertConversation(
+            newConversationEntity(
+                id = QualifiedIDEntity("1", "someDomain"),
+                lastReadDate = "2000-01-01T12:30:00.000Z",
+                lastModified = "2000-01-01T12:00:00.000Z"
+            )
+        )
+        conversationDAO.insertConversation(
+            newConversationEntity(
+                id = QualifiedIDEntity("2", "someDomain"),
+                lastReadDate = "2000-01-01T12:30:00.000Z",
+                lastModified = "2000-01-01T12:00:00.000Z"
+            )
+        )
+        conversationDAO.insertConversation(
+            newConversationEntity(
+                id = QualifiedIDEntity("3", "someDomain"),
+                lastReadDate = "2000-01-01T12:30:00.000Z",
+                lastModified = "2000-01-01T12:00:00.000Z"
+            )
+        )
+        conversationDAO.insertConversation(
+            newConversationEntity(
+                id = QualifiedIDEntity("3", "someDomain"),
+                lastReadDate = "2000-01-01T12:30:00.000Z",
+                lastModified = "2000-01-01T12:00:00.000Z"
+            )
+        )
+
+        // when
+        val result = conversationDAO.getUnreadConversationCount()
+
+        // then
+        assertEquals(0L, result)
+    }
+
+    @Test
+    fun givenMessagesArrivedAfterTheUserSawConversation_WhenGettingUnreadMessageCount_ThenReturnTheExpectedCount() = runTest {
+        // given
+        val conversationId = QualifiedIDEntity("1", "someDomain")
+
+        conversationDAO.insertConversation(
+            newConversationEntity(
+                id = conversationId,
+                lastReadDate = "2000-01-01T12:00:00.000Z",
+            )
+        )
+
+        userDAO.insertUser(user1)
+
+        val message = buildList {
+            // add 5 Message before the lastReadDate
+            repeat(5) {
+                add(
+                    newMessageEntity(
+                        id = it.toString(), date = "2000-01-01T12:0$it:00.000Z",
+                        conversationId = conversationId,
+                        senderUserId = user1.id,
+                    )
+                )
+            }
+            // add 5 Message past the lastReadDate
+            repeat(5) {
+                add(
+                    newMessageEntity(
+                        id = it.toString(), date = "2000-01-01T12:1$it:00.000Z",
+                        conversationId = conversationId,
+                        senderUserId = user1.id,
+                    )
+                )
+            }
```

Comment:
This test is buggy.

The IDs are clashing and messages are being overwritten.
```kt
repeat(5){
    newMessageEntity(id = it.toString())
}
// IDs = 0, 1, 2, 3 and 4
repeat(5){
    newMessageEntity(id = it.toString())
}
// Same IDs = 0, 1, 2, 3 and 4
// So the messages are overwriting the ones added before when inserting into DB
```


Also, the dates of the `before the lastReadDate` are all equal or past the lastReadDate:

`lastReadDate = "2000-01-01T12:00:00.000Z"`

We're only cahnging after T12:
`2000-01-01T12:0$it:00.000Z`

So:
```kt
repeat(5){
    println("12:0$it:00.000Z") 
}
// 12:00:00.000Z (same as last read date)
// 12:01:00.000Z (after last read date)
// 12:02:00.000Z (after last read date)
// 12:03:00.000Z (after last read date)
// 12:04:00.000Z (after last read date)
```

And then, for the `past the lastReadDate` we're doing the same, but starting with `12:1`


I suggest using iterating the seconds and changing the minutes instead:

```kt
// 12:10
lastReadDate = "2000-01-01T12:10:00.000Z"


// Before
repeat(5) {
    // 12:00
    println("2000-01-01T12:00:0$it.000Z")
}

// After
repeat(5) {
    // 12:20
    println("2000-01-01T12:20:0$it.000Z")
}
```